include/exadg/aero_acoustic/single_field_solvers/acoustics.h

```diff
@@ -80,6 +117,26 @@ class SolverAcoustic
 
   // postprocessor
   std::shared_ptr<Acoustics::PostProcessorBase<dim, Number>> postprocessor;
+
+private:
+  double
+  compute_sub_time_step_size(double const global_dt)
+  {
+    double const current_dt = time_integrator->get_time_step_size();
+    if(global_dt < current_dt)
+    {
+      // in case the global time step is smaller than the acoustic timestep, simply run
+      // with global time step
+      return global_dt;
+    }
+    else
+    {
+      // otherwise, ensure that sub-timestepping ends exactly at global_dt
+      return adjust_time_step_to_hit_end_time(0.0, global_dt, current_dt);
+    }
+  }
+
+  std::pair<unsigned int /* calls */, unsigned long long /* sub_dt */> sub_time_steps;
```

Comment:
Related to comment https://github.com/exadg/exadg/pull/645/files#r1561011435, "calls" and "sub_dt" looks somewhat misleading to me. What about "number of macro time steps" and "total number of sub time steps"?

Reply:
[e4a3c21](https://github.com/exadg/exadg/pull/645/commits/e4a3c212a070b3a5001a68fb3583f88667ec37dc)